components/server/src/github/github-context-parser.ts

```diff
@@ -416,4 +416,56 @@ export class GithubContextParser extends AbstractContextParser implements IConte
             }` : ''}
         `;
     }
+
+    public async fetchCommitHistory(ctx: TraceContext, user: User, contextUrl: string, sha: string, maxDepth: number): Promise<string[]> {
+        const span = TraceContext.startSpan("GithubContextParser.fetchCommitHistory", ctx);
+
+        try {
+            if (sha.length != 40) {
+                throw new Error(`Invalid commit ID ${sha}.`);
+            }
+
+            // FIXME: To get more results than GitHub API's max page size (seems to be 100), pagination should be handled.
```

Comment:
```suggestion
            // TODO(janx): To get more results than GitHub API's max page size (seems to be 100), pagination should be handled.
```

Reply:
see https://www.notion.so/gitpod/How-we-develop-e8ce7e562478458a9223eb9b797415b2#86e669f8e7df4f1289c9bede0855b5f9